component/src/main/java/org/ballerinalang/sap/nativeimpl/consumer/Stop.java

```diff
@@ -0,0 +1,130 @@
+            type = TypeKind.OBJECT,
+            structType = SapConstants.CONSUMER_STRUCT_NAME,
+            structPackage = SapConstants.SAP_NATIVE_PACKAGE
+    ),
+    isPublic = true
+)
+public class Stop implements NativeCallableUnit {
+    private static final Log log = LogFactory.getLog(Stop.class);
+    @Override
+    public void execute(Context context, CallableUnitCallback callableUnitCallback) {
+        Struct consumerStruct = BLangConnectorSPIUtil.getConnectorEndpointStruct(context);
+        String transportName = (String) consumerStruct.getNativeData(CONSUMER_TRANSPORT_NAME);
+        String serverName = (String) consumerStruct.getNativeData(CONSUMER_SERVER_STRUCT_NAME);
+
+        if (transportName.equalsIgnoreCase(SAP_IDOC_PROTOCOL_NAME)) {
+            JCoIDocServer jcoIDocServer = (JCoIDocServer) consumerStruct.getNativeData(CONSUMER_SERVER_CONNECTOR_NAME);
+
+            if (log.isDebugEnabled()) {
+                log.debug("Stopping the JCo endpoint : " + serverName);
+            }
+            jcoIDocServer.stop();
+            jcoIDocServer.release();
+
+            if (waitForServerStop(jcoIDocServer)) {
+                log.warn("JCo server : " + serverName + " is taking an unusually long time to stop.");
+            } else {
+                log.info("JCo server : " + serverName + " stopped");
+            }
+        } else if (transportName.equalsIgnoreCase(SAP_BAPI_PROTOCOL_NAME)) {
+            JCoServer jcoServer = (JCoServer) consumerStruct.getNativeData(CONSUMER_SERVER_CONNECTOR_NAME);
+
+            if (log.isDebugEnabled()) {
+                log.debug("Stopping the JCo endpoint : " + serverName);
+            }
+            jcoServer.stop();
+            jcoServer.release();
+
+            if (waitForServerStop(jcoServer)) {
+                log.warn("JCo server : " + serverName + " is taking an unusually long time to stop.");
+            } else {
+                log.info("JCo server : " + serverName + " stopped");
+            }
+        } else {
+            throw new BallerinaException("Protocol name: " + transportName + " is not " +
+                    "supported.");
+        }
+        context.setReturnValues();
+    }
+
+    /**
+     * Block until server state is stopped or maximum timeout reached.
+     *
+     * @param jcoServer jco server for which we wait to be stopped.
+     * @return true if the server is stopped before the timeout is exceeded
+     */
+    private boolean waitForServerStop(JCoServer jcoServer) {
+        long timeStamp = System.currentTimeMillis();
+        while (jcoServer.getState() != JCoServerState.STOPPED
+                && timeStamp + serverStopTimeout > System.currentTimeMillis()) {
+            if (log.isDebugEnabled()) {
+                log.debug("Waiting for server to stop...");
+            }
+            try {
+                TimeUnit.SECONDS.sleep(1);
+            } catch (InterruptedException e) {
+                //do nothing, just continue the loop
```

Comment:
```suggestion
                //Continue the loop.
```

Can we say skip this section?